gramps/gen/filters/_genericfilter.py

```diff
@@ -144,51 +147,34 @@ def apply_logical_op_to_all(
     ):
         final_list = []
 
-        optimizer = Optimizer(self)
-        handles_in, handles_out = optimizer.get_handles()
+        if id_list:
+            optimizer = Optimizer(set(id_list), self)
+        else:
+            optimizer = Optimizer(set(self.get_all_handles(db)), self)
```

Comment:
```suggestion
        optimizer = Optimizer(set(id_list if id_list else self.get_all_handles(db)), self) 
```
This might be cleaner

Reply:
There's more to it than this. I'll add a commit

Reply:
Nice catch!